telegram/ext/_application.py

```diff
@@ -738,33 +740,31 @@ async def __process_update_wrapper(self, update: object) -> None:
             self.update_queue.task_done()
 
     async def process_update(self, update: object) -> None:
-        """Processes a single update and updates the persistence.
+        """Processes a single update and marks the update to be updated by the persistence later.
 
         .. versionchanged:: 14.0
-            This calls :meth:`update_persistence` exactly once after handling of the update was
-            finished by *all* handlers that handled the update, including asynchronously running
-            handlers.
+            Persistence is now updated in an interval set by
+            :attr:`telegram.ext.BasePersistence.update_interval`.
 
         Args:
             update (:class:`telegram.Update` | :obj:`object` | \
-                :class:`telegram.error.TelegramError`):
-                The update to process.
+                :class:`telegram.error.TelegramError`): The update to process.
 
         """
         context = None
-        any_blocking = False
+        any_blocking = False  # Flag which is set to True if any handler specifies block=True
 
         for handlers in self.handlers.values():
             try:
                 for handler in handlers:
-                    check = handler.check_update(update)
-                    if not (check is None or check is False):
-                        if not context:
+                    check = handler.check_update(update)  # Should the handler handle this update?
+                    if not (check is None or check is False):  # if yes,
+                        if not context:  # build a context if not already built
```

Comment:
IMO this is a bit overly verbose/close to stating the obvious … just my 2cts - you can keep it if you like.

Reply:
I know it's obvious to us, but not sure if it is obvious to someone new / unfamiliar to PTB